src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1508,31 +1534,61 @@ public Text getUsage(CommandSource src) {
     }
 
     public static CommandElement entity(Text key) {
-        return new EntityCommandElement(key, false);
+        return new EntityCommandElement(key, false, false, null);
     }
 
     public static CommandElement entityOrSource(Text key) {
-        return new EntityCommandElement(key, true);
+        return new EntityCommandElement(key, true, false, null);
+    }
+
+    public static CommandElement entity(Text key, Class<? extends Entity> type) {
```

Comment:
I appended to the bottom; if this is an issue I'll move it.

Intentional; if `orSource` is used, the target type is a player.